src/native/_base.js

```diff
@@ -74,7 +74,56 @@ module.exports = {
       d.setTime((sec * 1000) + (nsec_part / 1000000));
       return d;
     },
-    Promisified: function(formatter, rTypes, after, isLastArgCb) {
+    Promisified: function(formatter, rTypes, after) {
+      // create internal function that will be
+      // invoked ontop of the direct binding
+      // mixing a callback into the arguments
+      // and returning a promise
+      return (lib, fn) => (function() {
+        // the internal function that wraps the actual function call
+        // compile the callback-types-definiton
+        let args;
+        let types = ['pointer', FfiResult]; // we always have: user_context and FfiResult
+        if (Array.isArray(rTypes)) {
```

Comment:
Can be refactored using the spread operator
types.push(Array.isArray(rTypes) ? [...rTypes] : [rTypes]);

Reply:
I guess you meant `types.concat(...)` since you cannot push an array but concatenate it, but that still doesn't work since in some cases `rTypes` can be `null`.
I just pushed a new commit where I use the spread operator but just to replace the `concat` call.